java-client/session/src/main/java/io/deephaven/client/impl/SessionImpl.java

```diff
@@ -282,26 +278,10 @@ public CompletableFuture<Void> closeFuture() {
     }
 
     @Override
-    protected TableHandleManager delegate() {
-        return config.delegateToBatch() ? batchManager : serialManager;
-    }
-
-    @Override
-    public TableHandleManager batch() {
-        return batchManager;
-    }
-
-    @Override
-    public TableHandleManager batch(boolean mixinStacktrace) {
-        if (this.config.mixinStacktrace() == mixinStacktrace) {
-            return batchManager;
-        }
-        return TableHandleManagerBatch.of(this, mixinStacktrace);
-    }
-
-    @Override
-    public TableHandleManager serial() {
-        return serialManager;
+    protected TableServices delegate() {
```

Comment:
So the pattern is we construct a whole bunch of state, and then throw it away...

Reply:
Yes. The internal state _is_ needed even for a single query - there may be shared sub-DAGs that we don't want to export multiple times in a single Batch request. One might argue "we should re-implement all of this logic, only keeping the state/Map alive for the duration of the execute method", but re-working  as such would add additional maintenance burdens for the sake of saving the construction of a extra object, which is not even close to worth it IMO.